libkbfs/block_retrieval_queue.go

```diff
@@ -390,22 +404,40 @@ func (brq *blockRetrievalQueue) request(ctx context.Context,
 			}
 		}
 	}
+	// Update the action if we receive a stronger follow-on action.
+	if action > br.action {
+		br.action = action
+	}
 	return ch
 }
 
 // Request implements the BlockRetriever interface for blockRetrievalQueue.
 func (brq *blockRetrievalQueue) Request(ctx context.Context,
 	priority int, kmd KeyMetadata, ptr BlockPointer, block Block,
 	lifetime BlockCacheLifetime) <-chan error {
-	return brq.request(ctx, priority, kmd, ptr, block, lifetime, true)
+	action := blockRequestWithPrefetch
```

Comment:
I did it this way because at first I was passing around booleans and I wanted to keep the interface clean.  But then I switched to explicit actions, so I could expose that in the interface, but I still think this way is a little cleaner and allows for separate interface method comments explaining what's going on.  But I'm happy to go back to a single `Request` that takes an explicit action, if you think that's better.

Reply:
Actually, I'm going to go with your suggestion now, because I was thinking about the next phase of partial syncing and I think we'll need a fourth type of action for syncing without prefetching.  I'll have to export the actions, but I think it makes sense to collapse it into a single request function now.